k8s_helm_charts/zdm/templates/service.yaml

```diff
@@ -21,6 +21,6 @@ spec:
       name: cql
   selector:
     {{- $zdm_selectorLabels | nindent 4 }}
-    app: {{ $zdm_fullname }}-{{ $index }}
+    statefulset.kubernetes.io/pod-name: {{ $zdm_fullname }}-{{ $index }}
```

Comment:
Why are you trying to match a per-pod label?

Reply:
@jsanda The reason for us to create individual services and match them 1:1 at pod level is because unlike C*, zdm-proxy processes don't have gossip protocol to detect/notify topology changes among themselves; whatever set of IP addresses and their respective ordinal index (see [here](https://github.com/datastax/zdm-proxy/blob/f893984498547d5181c0216e955e01949ff5b88b/proxy/pkg/config/config.go#L27-L28)) specified at the start of the proxy process is going to stuck for the rest of its lifecycle. To allow dynamical IP address change to the pods due to reschedule/cordon/crash, we decided to for N number of proxies, implement N number of services, and them map 1-on-1. Given that pods managed by k8s Deployment object doesn't have a static pod name, we switched to using StatefulSets. In StatefulSets, each pod has a unique/static pod-name that we can map the service to. This will allow orderly rolling restart of the zdm-proxy pods to happen.